test-framework/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/core/condition/MultipleWaitConditionHandler.java

```diff
@@ -30,6 +30,13 @@
 	
 	private Map<WaitCondition, Consumer<Object>> awaitsConditions = new HashMap<WaitCondition, Consumer<Object>>();
 
+	public MultipleWaitConditionHandler(Map<WaitCondition, Consumer<Object>> conds) {
+		awaitsConditions = conds;
+		for (WaitCondition wait : conds.keySet()) {
+			description += wait.description() + " | ";
```

Comment:
@odockal dare to address the sonar complaint? Seems legit since "+" on strings is compiled to StringBuilder calls, so in your case your repeatedly creating a new StringBuilder and appending to it.